src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerExecDecorator.java

```diff
@@ -417,19 +418,31 @@ public void onClose(int i, String s) {
                         });
 
                 ExecWatch watch;
-                try {
-                    watch = execable.exec(sh);
-                } catch (KubernetesClientException e) {
-                    if (e.getCause() instanceof InterruptedException) {
+                int retryCount = 0;
+                int maxTries = 3;
+                while(true) {
+                    try {
+                        watch = execable.exec(sh);
+                        break;
+                    } catch (KubernetesClientException e) {
+                        if (e.getCause() instanceof InterruptedException) {
+                            throw new IOException(
+                                    "Interrupted while starting websocket connection, you should increase the Max connections to Kubernetes API",
+                                    e);
+                        } else if (e.getCause() instanceof ProxyException && e.getMessage().contains("container not found")) {
+                            if (++retryCount == maxTries) {
+                                throw e;
+                            } else {
+                                LOGGER.log(Level.WARNING,
+                                        "Caught ProxyException(container not found), retrying exec command...");
```

Comment:
do you feel waiting (with some backoff?) may be worth? I guess it depends on the root issue / for how long we may be hitting it..